DOCS-3172: Change references to old module generator to CLI (#3745)
viamrobotics/docs: docs/how-tos/sensor-module.md

1. [Start with a test script](#start-with-a-test-script)
1. [Generate template module code](#generate-template-module-code)
1. [Implement the sensor API](#implement-the-sensor-api)
1. [Test your module locally](#test-your-module-locally)
1. [Upload your module](#upload-your-module-to-the-registry)

{{% /alert %}}

## Prerequisites

{{< expand "Install the Viam CLI and authenticate" >}}

> Install the Viam CLI and authenticate to Viam, from the same machine that you intend to upload your module from.

Reviewer: Authenticate is the next bit here, so probably cleaner to put them one after the other
```suggestion
Install the Viam CLI on the same machine that you intend to upload your module from.
```

There are a few standardized files that must be part of any module.
You can create these automatically using the Viam module generator:

1. Run the `module generate` command in your terminal:

   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
   viam module generate
   ```

2. Follow the prompts, naming your module and selecting from the options.

<!--prettier-ignore-->
| Prompt | Description |
| -------| ----------- |

> | Module name | The module name (also called repo name or family name) is generally the name of the GitHub repo where you will put your module code. Name it something related to what your module does. For example, `weather`. |

Reviewer: It'd not also called repo name.
```suggestion
| Module name | The module name describes the module or the family of devices it supports. It is generally the same as the name of the GitHub repo where you will put your module code. For example, `weather`. |
```

Author: Okay. This is how it was previously referred to but I guess we can guide language here :)

Author: "Repo name" is indeed a confusing thing to call it and only some module repo names match module names--we should fix elsewhere too: https://viam.atlassian.net/browse/DOCS-3236

> | Language | Choose Python to follow this tutorial. |

Author: Oop typed the wrong word

| Visibility | Choose `Private` to share only with your organization, or `Public` to share publicly with all organizations. If you are testing, choose `Private`. |
| Namespace/Organization ID | In the [Viam app](https://app.viam.com), navigate to your organization settings through the menu in upper right corner of the page. Find the **Public namespace** and copy that string. In the example snippets below, the namespace is `jessamy`. |

> | Resource to add to the module (API) | The [component API](/appendix/apis/#component-apis) or [service API](/appendix/apis/#service apis) the resource you're creating implements. Choose `Sensor Component` for this guide. |

Author: ```suggestion
| Resource to add to the module (API) | The [component API](/appendix/apis/#component-apis) or [service API](/appendix/apis/#service-apis) the resource you're creating implements. Choose `Sensor Component` for this guide. |
```

| Model name | Name your sensor based on what it supports, for example, if it supports a model of ultrasonic sensor called “XYZ Sensor 1234” you could call your model `XYZ_1234` or similar. |

> | Enable cloud build | You can select `No` for this tutorial because you'll build the module yourself before uploading it. If you select `Yes`, the module will build from your specified GitHub repo using GitHub actions. |

Reviewer: What specified GitHub repo? If you select yes it doesn't ask you to specify one and you don't need to specify one either. To get this to work you'd need to push it to a GitHub repo and create a release.

Author: That is correct. For the purposes of this guide I wanted to keep it simple while eluding to the other option.
```suggestion
| Enable cloud build | You can select `No` for this guide because you'll build the module yourself before uploading it. If you select `Yes`, the module will use GitHub actions to build the module from a GitHub repo you create and specify in the <file>meta.json</file>. |
```

Reviewer: Right but you're alluding to it with wrong information that will potentially confuse the reader. The generated .github folder is what will make github automatically use actions once the files are pushed to github and a release is created (which is what I said above). It works like our github actions defined in our .github folder in the docs. I don't think you'd need to specify a GitHub repo in the meta.json at all. The fact that the generator doesn't ask you to and doesn't generate a placeholder for it is an issue because that means our registry won't be able to link to the repo but that is unrelated. It is feedback we should discuss with the CLI team though.

```suggestion
| Enable cloud build | You can select `No` for this tutorial because you'll build the module yourself before uploading it. If you select `Yes` and push the generated files (including the <FILE>.github</FILE> folder) and create a release of the format `vX.X.X`, the module will build and upload to the Viam registry. |
```

Author: Ah so it's pushed from the repo, not pulled by the registry. I see. 
(Changing to "guide")
```suggestion
| Enable cloud build | You can select `No` for this guide because you'll build the module yourself before uploading it. If you select `Yes` and push the generated files (including the <file>.github</file> folder) and create a release of the format `vX.X.X`, the module will build and upload to the Viam registry. |
```

> | Register module | `Yes` unless you are creating a local-only module for testing purposes and do not intend to upload it. |

Reviewer: ```suggestion
| Register module | Select `Yes` unless you are creating a local-only module for testing purposes and do not intend to upload it. |
```

> The generator will generate a folder containing stub files for your modular sensor component.

Reviewer: ```suggestion
The generator will create a folder containing stub files for your modular sensor component.
```

Update the generated <file>requirements.txt</file> file to include any packages that must be installed for the module to run.
Depending on your use case, you may not need to add anything here beyond `viam-sdk` which is auto-populated.

> ## Make the module executable

Reviewer: good call
